Doc 12584 new javascript plugin doc

ElliotFrancisHunter opened the pull request:
PR to update the Javascript doc with the new ionic and react plugins created by the DevRel team. 

Copy doc here: https://docs.google.com/document/d/1i7RQDm9IP4vGEVLXsu1FVX7y2V_Dt5z9OQj2x51XiFM/edit?pli=1

Preview Credentials: https://couchbasecloud.atlassian.net/wiki/spaces/DOCS/pages/1971224949/Docs+Team+demo+site+passwords

Changed Pages
-------------------

- What's new page: https://preview.docs-test.couchbase.com/jsplugins/couchbase-lite/current/cbl-whatsnew.html
- JavaScript quick-start page: https://preview.docs-test.couchbase.com/jsplugins/couchbase-lite/current/javascript/quickstart.html
- Ionic Page: https://preview.docs-test.couchbase.com/jsplugins/couchbase-lite/current/javascript/ionic.html
- React Page: https://preview.docs-test.couchbase.com/jsplugins/couchbase-lite/current/javascript/react.html

----------------------

Pages still needing changes/updates
-----------------------

....

-----------------------

ElliotFrancisHunter committed:
DOC-12584: Dead file removal

ElliotFrancisHunter committed:
DOC-12584: What's new page addition

ElliotFrancisHunter committed:
DOC-12584: Move Cordova down

Move Cordova to bottom of nav as it's not being actively maintained.

ElliotFrancisHunter committed:
DOC-12584: What's new addition

Added plugin announcement to what's new page.

ElliotFrancisHunter committed:
DOC-12584: Update JavaScript quickstart page.

Changelist:
-- Removing 404 ionic link
-- Updating Ionic and React sections to include new plugin documentation links and the new react example project

ElliotFrancisHunter committed:
DOC-12584: Update and replace existing documentation

Replaced existing doc with new content for the ionic and react doc pages

ElliotFrancisHunter commented:
@biozal @idulo some questions I have on this documentation in general:

- Are we axing the existing Ionic/React doc to replace with this new information? That's what I've done currently.
- Is this a public preview or a beta? I've seen both being used and I'm not sure which one is which.
- Should I remove the ionic example from the quickstart page? I believe it's for the old project.
- Are we keeping the cordova documentation or should that just be removed so we only have React and Ionic?
- What timeline are we looking at for GA? Are we doing a 3.2 public preview or is it previewing on 3.1.x to preview on 3.2.x? I've just documented it in the 3.2 docs for now.

biozal commented:
It's a public beta but since it's open source there is very little meaning to beta vs final release outside we know there are things that aren't working properly which is why we are calling them beta.

I highly recommend axing the current versions of the pages we have and any quickstarts that point to the old stuff, it's only going to confuse people.  GA is a moving target but the thought pattern is Ionic is sometime in October/November and React Native is sometime in Novembver/December.  

We are working on learning paths for both Ionic and React Native that will be on the developer portal.

ElliotFrancisHunter committed:
DOC-12584: Remove cordova plugin content from docs

Remove doc page, nav entry and quickstart links

ElliotFrancisHunter committed:
DOC-12584: Doc feedback

Remove features and limitations from react as already in devrel doc
Add support information
Add missing links

RayOffiah commented on modules/javascript/pages/ionic.adoc:
```diff
@@ -1,77 +1,34 @@
 
-// end::summary[]
-This integration supports apps built for iOS, Android, and native Windows.
+NOTE: Couchbase Community Supported is an Open source integration that is developed by Couchbase. Issues can also be raised in the relevant GitHub repositories or via Couchbase forums or Discord. Support is provided best effort by development team.
```
Is `Couchbase Community Supported` an `integration` or is it a support scheme?

ElliotFrancisHunter replied:
I believe it's a support scheme, `Couchbase Community Supported` is the term for an open source integration with best effort dev support on our end.

RayOffiah commented on modules/javascript/pages/react.adoc:
```diff
@@ -1,56 +1,48 @@
-:param-abstract: If you are building mobile or desktop applications using web technologies such as JavaScript/HTML/CSS, there are options for using Couchbase Lite as an embedded database within your app.
-:param-related: xref:ionic.adoc[] | xref:cordova.adoc[]
-include::{root-partials}_show_page_header_block.adoc[]
+== High-Level Overview
```
I'd probably just say `Overview`

ElliotFrancisHunter replied:
Changed.

RayOffiah commented on modules/javascript/pages/react.adoc:
```diff
@@ -1,56 +1,48 @@
 
-NOTE: Couchbase Lite for React Native is a community-driven project without official support from Couchbase.
+* Community-Supported -  This open-source plugin is Couchbase community supported and maintained, and it requires an Enterprise Edition Couchbase Lite license. 
+While it is not officially supported by Couchbase, any issues specific to Couchbase Lite are covered under our existing support policy.
```
This sounds like it *is* supported.

rajagp replied:
@ElliotFrancisHunter  This section is not correct and it's quite confusing.  In general. please apply all the comments on Ionic page to the React Native page

ElliotFrancisHunter replied:
Done

RayOffiah reviewed (dismissed):
A bit unclear whether we support it or not, but that might be just me.

"We don't officially support it, but it's covered by other support licenses."

A bit unclear.

rajagp commented on modules/ROOT/pages/cbl-whatsnew.adoc:
```diff
@@ -17,6 +17,33 @@ NOTE: Couchbase Lite 3.0 introduces some breaking changes. +
 If you are upgrading from 2.x, please refer to the appropriate upgrade page -- see: <<lbl-upgrade>>
 Users should be able to upgrade to 3.1.x from 3.0.x without manual intervention.
 
+== New JavaScript Plugins for Couchbase Lite
```
@ElliotFrancisHunter  Please remove entire  Whats New Section related to plugins. We do not include release notes for couchbase community projects in our official docs and the plugin releases are not directly  linked to CBL releases. Any release notes will be part of that plugin documentation. 
Moot point but the the content is also wrong - The plugins are not enterprise

ElliotFrancisHunter replied:
@rajagp Done, I have some confusion here though. The plugins themselves require CBL with an enterprise license according to the doc so what differentiates requiring an enterprise version of CBL with being enterprise only?

rajagp commented on modules/javascript/pages/ionic.adoc:
```diff
@@ -1,77 +1,34 @@
+:page-edition: Public Preview
 :page-toclevels: 2@
-:description: Using Couchbase Lite with Javascript Ionic applications
+:description: Couchbase is excited to announce the public preview of the Ionic Capacitor plugin for Couchbase Lite, which allows developers to leverage Couchbase Lite within cross-platform applications in JavaScript for iOS and Android. This plugin is open-source and supported by the Couchbase community. 
```
```suggestion
:description:  The Ionic Capacitor plugin for Couchbase Lite, allows developers to leverage Couchbase Lite within cross-platform applications in JavaScript for iOS and Android. 
```

ElliotFrancisHunter replied:
Added

ElliotFrancisHunter commented:
@biozal there's this ionic tutorial on the quickstart page: https://ionic.io/docs/couchbase-lite/tutorials/hotel-search

Should this be removed? Or will this page be updated to the new tutorial at some point?

rajagp commented on modules/javascript/pages/ionic.adoc:
```diff
@@ -1,77 +1,34 @@
 
-// tag::summary[]
-Ionic's Couchbase Lite integration is an open source solution for web developers, enabling easy to build secure, high-performance, offline-enabled apps.
+* Community-Supported - This open-source plugin is Couchbase community supported and maintained, and it requires an Enterprise Edition Couchbase Lite license.
```
```suggestion
* Couchbase Community-Supported - This open-source plugin is Couchbase Community supported. However, the Couchbase Lite binary that is linked to the plugin is Couchbase Lite EE and its use is guided by the T&C of the Couchbase Lite Enterprise license agreement < INCLUDE LINK> 
```

ElliotFrancisHunter replied:
Done

rajagp commented on modules/javascript/pages/ionic.adoc:
```diff
@@ -1,77 +1,34 @@
-// tag::summary[]
-Ionic's Couchbase Lite integration is an open source solution for web developers, enabling easy to build secure, high-performance, offline-enabled apps.
+* Community-Supported - This open-source plugin is Couchbase community supported and maintained, and it requires an Enterprise Edition Couchbase Lite license.
+While it is not officially supported by Couchbase, any issues specific to Couchbase Lite are covered under our existing support policy.
```
```suggestion
This implies that the use of the plugin in production applications requires an EE license of Couchbase Lite. So while the plugin itself is not officially supported by Couchbase, any issues specific to Couchbase Lite are covered under our existing support policy.


```

rajagp commented on modules/javascript/pages/ionic.adoc:
```diff
@@ -1,77 +1,34 @@
-https://ionic.io/integrations/couchbase-lite[landing page]
-
-****
+As this is a community-supported project, we encourage you to explore, test, and provide feedback to help us improve the React Native plugin for Couchbase Lite. Feel free to file issues through the https://github.com/Couchbase-Ecosystem/cbl-ionic/issues[GitHub issue tracker], our https://www.couchbase.com/blog/couchbase-on-discord/[Community Discord] or https://www.couchbase.com/forums/[Couchbase ...
```
```suggestion
As this is a community-supported project, we encourage you to explore, test, and provide feedback to help us improve the Ionic plugin for Couchbase Lite. Feel free to file issues through the https://github.com/Couchbase-Ecosystem/cbl-ionic/issues[GitHub issue tracker], our https://www.couchbase.com/blog/couchbase-on-discord/[Community Discord] or https://www.couchbase.com/forums/[Couchbase Forums].
```

ElliotFrancisHunter replied:
Good spot! Fixed

rajagp commented on modules/javascript/pages/ionic.adoc:
```diff
@@ -1,77 +1,34 @@
 
-// end::summary[]
-This integration supports apps built for iOS, Android, and native Windows.
+NOTE: Couchbase Community Supported is an Open source integration that is developed by Couchbase. Issues can also be raised in the relevant GitHub repositories or via Couchbase forums or Discord. Support is provided best effort by development team.
```
```suggestion
NOTE: _**Couchbase Community Supported**_ integration/plugin is an Open source integration that is developed by Couchbase. Issues related to the integration can also be raised in the relevant GitHub repositories or via Couchbase forums or Discord. Support is provided best effort by development team.
```

ElliotFrancisHunter replied:
Added

rajagp reviewed (changes requested):
Please apply comments on Ionic to React Native page as well

ElliotFrancisHunter committed:
DOC-12584: Dead code removal

ElliotFrancisHunter committed:
DOC-12584: Applying PM and docs feedback

osfameron commented on modules/javascript/pages/quickstart.adoc:
```diff
@@ -36,22 +36,11 @@ image::ROOT:manage-and-store-data-locally.svg[,250]
-====== {empty}
-[.content]
-.xref:javascript:cordova.adoc[Cordova]
+* https://cbl-ionic.dev[Plugin{nbsp}Documentation]
```
Not sure if you need the {nbsp} here - I realise it's copied from the existing links so not a biggie, just a possible opportunity for cleanup 🤷

ElliotFrancisHunter replied:
Tested without and seems fine 🤷🏼 , removed!

osfameron commented on modules/javascript/pages/quickstart.adoc:
```diff
@@ -88,12 +81,6 @@ image::ROOT:manage-and-store-data-locally.svg[,250]
-
-
-
 // DO NOT EDIT OR REMOVE
```
go on, I dare you.

ElliotFrancisHunter replied:
🤣 done!

osfameron commented on modules/javascript/pages/react.adoc:
```diff
@@ -1,56 +1,48 @@
+Note that the Couchbase Lite binary linked to the plugin is Couchbase Lite Enterprise Edition (EE) and its use is guided by the terms and conditions of the https://www.couchbase.com/2018-04-30v3_License_Agreement/[Couchbase Lite Enterprise license agreement].
 
-// == Introduction
+NOTE: _**Couchbase Community Supported**_ is an Open source integration that is developed by Couchbase. Issues related to the integration can be raised in the relevant GitHub repositories or via Couchbase forums or Discord. Support is provided best effort by development team.
```
bold AND italic? 😱

osfameron replied:
also did you mean the name of the plugin?
And careful on "Community Supported" vs "Community-Supported" consistency.

ElliotFrancisHunter replied:
Removed hyphenation. Keeping the bold AND italic for now as was suggested by PM and is clear/legible in the local docs build.

osfameron reviewed (approved):
Looks good - couple of inline comments.

simon-dew commented on modules/javascript/pages/quickstart.adoc:
```diff
@@ -62,6 +51,10 @@ image::ROOT:manage-and-store-data-locally.svg[,250]
 
 * https://reactnative.dev/docs/native-modules-intro[React{nbsp}Native{nbsp}Modules]
 
+* https://cbl-reactnative.dev[Plugin{nbsp}Documentation]
```
Likewise with the `{nbsp}`, is it necessary?

simon-dew replied:
Looks like this was fixed already!

ElliotFrancisHunter committed:
DOC-12584: Docs feedback tidy up

simon-dew commented on modules/javascript/pages/ionic.adoc:
```diff
@@ -1,77 +1,34 @@
 
-.Support Changes
-[IMPORTANT]
+NOTE: Developers who are currently using the https://ionic.io/docs/couchbase-lite[previous version of the Ionic plugin] should migrate to the newer version of the plugin documented https://cbl-ionic.dev[here].
```
Not a big issue, but I think if a link is directing users to an external site, it would be polite to give them some clue as to where they're going. Something like

```suggestion
NOTE: Developers who are currently using the https://ionic.io/docs/couchbase-lite[previous version of the Ionic plugin] should migrate to the newer version of the plugin documented at https://cbl-ionic.dev[Couchbase Lite for Ionic Capacitor].
```

simon-dew commented on modules/javascript/pages/react.adoc:
```diff
@@ -1,56 +1,48 @@
+This public preview release enables developers using React to integrate Couchbase Lite into their mobile applications, ensuring they can build offline-first, cross-platform solutions with ease.
 
-:tutorial: https://github.com/couchbaselabs/userprofile-couchbase-mobile-reactnative/blob/main/README.md
+More information on React Native Native Modules can be found https://cbl-reactnative.dev/[here].
```
Likewise, external link, give the users a clue as to where they're headed

```suggestion
More information on React Native Native Modules can be found at https://cbl-reactnative.dev/[Couchbase Lite Enterprise Edition for React Native].
```

simon-dew commented on modules/javascript/pages/react.adoc:
```diff
@@ -1,56 +1,48 @@
-// == Functionality
+Couchbase recommends using Expo. 
+Expo makes working with React Native simpler and more efficient.
+You can find information on the prerequisites and setup, including Expo configuration https://cbl-reactnative.dev/StartHere/prerequisites[here].
```
Giving more of a clue for external links. This is just a suggestion, you could expand on this.

```suggestion
You can find information on the prerequisites and setup, including Expo configuration at https://cbl-reactnative.dev/StartHere/prerequisites[Prerequisites].
```

simon-dew reviewed (dismissed):
Generally seems okay, just some minor comments

ElliotFrancisHunter committed:
Apply suggestions from code review

Co-authored-by: Simon Dew <39966290+simon-dew@users.noreply.github.com>

ElliotFrancisHunter committed:
DOC-12584: More detailed link context

RayOffiah commented on modules/javascript/pages/react.adoc:
```diff
@@ -1,56 +1,48 @@
-include::{root-partials}_show_page_header_block.adoc[]
+== Overview
 
+Couchbase Lite Enterprise Edition for React Native is a Native Module implementation using TypeScript.
```
```suggestion
Couchbase Lite Enterprise Edition for React Native is a Native Module implementation that uses TypeScript.
```

RayOffiah reviewed (approved):
Tiny suggestion for a change, but nothing worth holding things up for.

ElliotFrancisHunter committed:
Update modules/javascript/pages/react.adoc

Co-authored-by: Ray Offiah <77050471+RayOffiah@users.noreply.github.com>
